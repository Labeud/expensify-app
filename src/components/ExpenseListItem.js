import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";
import 'numeral/locales/fr';

numeral.locale("fr");

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <Link to={`edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>
      {numeral(amount / 100).format("0,0.00$")}
      - 
      {moment(createdAt).format("DD/MM/YYYY")}
    </p>
  </div>
);

// No need to access the state, we just want dispatch
export default ExpenseListItem;