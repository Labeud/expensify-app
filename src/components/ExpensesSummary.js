import React from "react";
import { connect } from "react-redux";
import getTotalExpensesAmount from "../selectors/expensesTotal";
import selectExpenses from "../selectors/expenses";
import numeral from "numeral";
import 'numeral/locales/fr';
numeral.locale("fr");

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
  const expenseWord = expensesCount > 1 ? "expenses" : "expense";
  const formattedAmount = numeral(expensesTotal / 100).format("0,0.00$");
  return (
    <div>
      <h2>
        Viewing {expensesCount} {expenseWord} totalling {formattedAmount}.
      </h2>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expensesTotal: getTotalExpensesAmount(visibleExpenses),
    expensesCount: visibleExpenses.length
  };
};

export default connect(mapStateToProps)(ExpensesSummary);