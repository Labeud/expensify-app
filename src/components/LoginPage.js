import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";

export const LoginPage = ({ startLogin }) => (
  <div>
    <h1>Welcome to the Expensify App!</h1>
    <h3>Please login</h3>
    <button onClick={startLogin}>Login</button>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);