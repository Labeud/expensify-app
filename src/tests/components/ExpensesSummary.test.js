import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("should render correctly ExpensesSummary with 1 expense", () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={2000} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render correctly ExpensesSummary multiple expenses", () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={3} expensesTotal={1000} />);
  expect(wrapper).toMatchSnapshot();
});