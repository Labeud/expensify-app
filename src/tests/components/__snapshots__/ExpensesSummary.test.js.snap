// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly ExpensesSummary multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    3
     
    expenses
     totalling 
    10,00€
    .
  </h2>
</div>
`;

exports[`should render correctly ExpensesSummary with 1 expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    20,00€
    .
  </h2>
</div>
`;
